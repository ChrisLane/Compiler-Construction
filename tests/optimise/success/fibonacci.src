// This test will evaluate to print fibonacci numbers less than 50.
main () {
    var x = 50;
    var a = 0;
    var b = 1;

    print $a;
    while ($b <= ($x-1)) {
        print $b;
        var temp = $a + $b;
        a = $b;
        b = $temp;
    };
}
