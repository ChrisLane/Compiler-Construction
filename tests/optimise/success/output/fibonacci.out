
main (  ) { 
    New ( "x" = Const 50 ) In { 
        New ( "a" = Const 0 ) In { 
            New ( "b" = Const 1 ) In { 
                Print ( Const 0 ) ; 
                Print ( Const 1 ) ; 
                New ( "temp" = Const 2 ) In { 
                    Asg ( Identifier "a" := Const 1 ) ; 
                    Asg ( Identifier "b" := Const 1 ) 
                } ; 
                Print ( Const 1 ) ; 
                New ( "temp" = Const 3 ) In { 
                    Asg ( Identifier "a" := Const 1 ) ; 
                    Asg ( Identifier "b" := Const 2 ) 
                } ; 
                Print ( Const 2 ) ; 
                New ( "temp" = Const 5 ) In { 
                    Asg ( Identifier "a" := Const 2 ) ; 
                    Asg ( Identifier "b" := Const 3 ) 
                } ; 
                Print ( Const 3 ) ; 
                New ( "temp" = Const 8 ) In { 
                    Asg ( Identifier "a" := Const 3 ) ; 
                    Asg ( Identifier "b" := Const 5 ) 
                } ; 
                Print ( Const 5 ) ; 
                New ( "temp" = Const 13 ) In { 
                    Asg ( Identifier "a" := Const 5 ) ; 
                    Asg ( Identifier "b" := Const 8 ) 
                } ; 
                Print ( Const 8 ) ; 
                New ( "temp" = Const 21 ) In { 
                    Asg ( Identifier "a" := Const 8 ) ; 
                    Asg ( Identifier "b" := Const 13 ) 
                } ; 
                Print ( Const 13 ) ; 
                New ( "temp" = Const 34 ) In { 
                    Asg ( Identifier "a" := Const 13 ) ; 
                    Asg ( Identifier "b" := Const 21 ) 
                } ; 
                Print ( Const 21 ) ; 
                New ( "temp" = Const 55 ) In { 
                    Asg ( Identifier "a" := Const 21 ) ; 
                    Asg ( Identifier "b" := Const 34 ) 
                } ; 
                Print ( Const 34 ) ; 
                New ( "temp" = Const 89 ) In { 
                    Asg ( Identifier "a" := Const 34 ) ; 
                    Asg ( Identifier "b" := Const 55 ) 
                } ; 
                empty 
            } 
        } 
    } 
}

